File: 02-conversation-webhook/python/outbound_call.py
import os

from dotenv import load_dotenv
from phonic import Phonic

load_dotenv(".env.local")

client = Phonic(api_key=os.getenv("PHONIC_API_KEY"))


def outbound_call():
    client.conversations.outbound_call(
        to_phone_number=os.getenv("CUSTOMER_PHONE_NUMBER"),
        config={
            "agent": "travel-agent",
            "template_variables": {
                "customer_name": "Alice",
                "interest": "nature",
            },
        },
    )


if __name__ == "__main__":
    outbound_call()

File: 02-conversation-webhook/node/outbound-call.ts
import { config } from "dotenv";
import { PhonicClient } from "phonic";

config({ path: ".env.local" });

const client = new PhonicClient({
  apiKey: process.env.PHONIC_API_KEY,
  baseUrl: process.env.PHONIC_API_URL,
});

async function outboundCall() {
  const conversation = await client.conversations.outboundCall({
    to_phone_number: process.env.CUSTOMER_PHONE_NUMBER as string,
    config: {
      agent: "travel-agent",
      template_variables: {
        customer_name: "Alice",
        interest: "nature",
      },
    },
  });

  console.log("Outbound conversation:", conversation);
}

outboundCall();
